src/sentry/workflow_engine/processors/workflow_preview.py

```diff
@@ -0,0 +1,114 @@
+import logging
+from dataclasses import dataclass
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.models.environment import Environment
+from sentry.models.group import Group
+from sentry.workflow_engine.models import DataCondition, DataConditionGroup, Detector
+from sentry.workflow_engine.processors.data_condition import split_conditions_by_speed
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class EvaluationGroup:
+    logic_type: DataConditionGroup.Type
+    conditions: list[DataCondition]
+
+
+def preview_conditions(
+    logic_type: DataConditionGroup.Type, conditions: list[DataCondition], group_ids: set[int]
+) -> tuple[set[int], set[int]]:
+    if not conditions:
+        # no conditions, all groups are valid
+        return set(), group_ids
+
+    # groups that would have triggered each condition
+    conditions_to_triggered_groups = {}
+
+    for condition in conditions:
+        conditions_to_triggered_groups[condition.id] = condition.get_preview_groups(group_ids)
```

Comment:
How will we get the preview conditions here? Is each condition going to need to do some querying?

Reply:
yes they will do their own querying and return the group_ids that meet the condition

Reply:
This might end up being a bit more inefficient compared to the old implementation, since it uses activity and I think most conditions are able to infer what happened from that.

I guess in most cases, we won't have a lot of conditions, and we can make batch queries within the conditions as needed

Reply:
the handlers can use their choice of whatever filtering they need to do with the group id to filter correctly -- some use Activity and others filter on Group

Reply:
i was going to filter on `Activity` or `Group` just like the existing implementation, which also does a query per condition and pulls the `Activity` or `Group` result into memory via `values_list`